modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -106,18 +106,19 @@ class BaseConvolutionLayerImpl : public ConvolutionLayer
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
+        MatSize weightShape = blobs.empty() ? inputs[1].size : blobs[0].size;
         CV_Assert(inputs.size() > 0);
 
-        CV_Assert(blobs.size() == 1 || blobs.size() == 2);
+        CV_Assert((inputs.size() > outputs.size() && blobs.empty()) || blobs.size() == 1 || blobs.size() == 2);
```

Comment:
`inputs.size() > outputs.size()` seems strange

Reply:
if `blobs.empty()` then inputs contain input, weights and bias (optional) → ` inputs.size() > 1`, but `outputs.size() = 1`. I add this check instead of `inputs.size() > 1`, because in Deconvolution layer supports  `inputs.size() == outputs.size() >= 1.` https://github.com/opencv/opencv/blob/3.4/modules/dnn/src/layers/convolution_layer.cpp#L1914

Reply:
Why not `>=` this way?

Reply:
Ok, I see. this way please modify a check `CV_Assert(inputs.size() > 0);` so it will go before 
```cpp
MatSize weightShape = blobs.empty() ? inputs[1].size : blobs[0].size;
```
and will consider both cases